karspexet/ticket/views.py

```diff
@@ -25,9 +26,9 @@
 
 
 def home(request):
-    upcoming_shows = Show.upcoming()
+    upcoming_shows = Show.upcoming().filter(visible=True).all()
 
-    return render(request, "home.html", {
+    return TemplateResponse(request, "home.html", {
```

Comment:
I'm curious to the difference between calling `render` and returning a `TemplateResponse` directly?

Reply:
For an actual request, there's no difference. It will automatically call `render` on the `TemplateResponse` object when needed. This mostly makes it easier (and a little bit faster) to test stuff.

Reply:
From [TemplateResponse and SimpleTemplateResponse](https://docs.djangoproject.com/en/1.11/ref/template-response/) in the django documentation:

> Standard HttpResponse objects are static structures. They are provided with a block of pre-rendered content at time of construction, and while that content can be modified, it isn’t in a form that makes it easy to perform modifications.
> 
> However, it can sometimes be beneficial to allow decorators or middleware to modify a response after it has been constructed by the view. For example, you may want to change the template that is used, or put additional data into the context.
> 
> TemplateResponse provides a way to do just that. Unlike basic HttpResponse objects, TemplateResponse objects retain the details of the template and context that was provided by the view to compute the response. The final output of the response is not computed until it is needed, later in the response process.